internal/security/setup/option/optionHelpers.go

```diff
@@ -214,25 +214,48 @@ func markComplete(dirPath string) error {
 	return nil
 }
 
-func getXdgRuntimeDir() string {
+func getWorkDir() string {
 	if xdgRuntimeDir, ok := os.LookupEnv(envXdgRuntimeDir); ok {
-		return xdgRuntimeDir
+		return filepath.Join(xdgRuntimeDir, pkiInitBaseDir)
+	} else if setup.Configuration.SecretsSetup.WorkDir != "" {
+		return setup.Configuration.SecretsSetup.WorkDir
 	}
-	// if $XDG_RUNTIME_DIR is undefined, default to /tmp
-	return defaultXdgRuntimeDir
+
+	// if not specified in config and $XDG_RUNTIME_DIR is undefined, default to /tmp
+	return filepath.Join(defaultWorkDir, pkiInitBaseDir)
+}
+
+func getCertConfigDir() (string, error) {
+	if setup.Configuration.SecretsSetup.CertConfigDir != "" {
+		return setup.Configuration.SecretsSetup.CertConfigDir, nil
+	}
+
+	return "", errors.New("Directory for certificate configuration files not configured")
 }
 
-func getPkiCacheDirEnv() string {
+func getCacheDir() string {
 	pkiCacheDir := os.Getenv(envPkiCache)
 	if pkiCacheDir == "" {
+		if setup.Configuration.SecretsSetup.CacheDir != "" {
+			return setup.Configuration.SecretsSetup.CacheDir
+		}
 		return defaultPkiCacheDir
 	}
 	return pkiCacheDir
 }
 
+func getDeployDir() string {
+	if setup.Configuration.SecretsSetup.DeployDir != "" {
+		return setup.Configuration.SecretsSetup.DeployDir
+	}
+
+	return defaultPkiDeployDir
```

Comment:
Perhaps you should also test that DepoyDir exists and is writable.

Reply:
+1 to add defensive checking existence and writable

Reply:
createDirectoryIfNotExists(dirName string)